packages/api/src/contracts/github.repository.ts

```diff
@@ -0,0 +1,9 @@
+import { RegisteredUser, User } from '@coderscamp/shared/models/user';
+
+export abstract class GithubRepositoryPort {
```

Comment:
Why we have 'abstract class' here? 
What about interface?

Reply:
I don't think we can. Nest needs a class to replace. Unless we just explicitly say in each module which repository it should use. 

https://github.com/CodersCrew/coderscamp/blob/c8830ad87213d435a8feb428ba0402208ac94b0e/packages/api/src/prisma/prisma.module.ts#L11-L15

Reply:
class is not required, you can use string or symbol (i prefer the second one): 

https://docs.nestjs.com/fundamentals/custom-providers

Example you can find here: 
https://github.com/CodersCrew/coders-board-v2/blob/master/backend/src/inviting-applicants/write-side/infrastructure/inviting-applicants-infrastructure.module.ts

with symbol APPLICANT_INVITATION_REPOSITORY